Pin Jasmine to compatible version

symbolist opened the pull request.

symbolist commented:
@rocioar This is ready for your review.

rocioar commented on .circleci/config.yml:
```diff
@@ -5,15 +5,6 @@ workflows:
-            -  build
-      - lib_unit_tests:
-            requires:
-            -  build
```
@symbolist why were this removed?

symbolist replied:
@rocioar The current Circle CI plan has only 25 build hours per month so temporarily disabled them while getting js tests working. Will remove this commit before merging.

rocioar commented on scripts/circle-ci-configuration.sh:
```diff
@@ -39,7 +39,7 @@ if [ -f $FIREFOX_FILE ]; then
    echo "File $FIREFOX_FILE found."
 else
    echo "Downloading firefox_45.0.2%2Bbuild1-0ubuntu1_amd64.deb."
```
We will have to update this comment too, it still says version 45.

symbolist commented:
@rocioar I have addressed your comments. But strangely the lms js suite is now failing with this error: https://openedx.atlassian.net/browse/LEARNER-2120

symbolist commented:
@rocioar Looks like it had to do with node_modules caching. I have added one more commit which changes the node modules restoration step to only do a restore if the `package.json` hash matches. So whenever `package.json` changes npm will install from scratch. Can you give this last commit a look?

rocioar commented:
@symbolist good to merge once you have squashed the commits

symbolist commented:
@rocioar The cherry-picked commits should maintain their provenance. The other two are also independent changes so leaving the commit structure as is. :)
